cmd/metal-api/internal/datastore/machine.go

```diff
@@ -429,6 +429,22 @@ func (rs *RethinkStore) ListMachines() (metal.Machines, error) {
 	return ms, err
 }
 
+func (rs *RethinkStore) listMachinesInPartition(partitionid, sizeid string) (metal.Machines, error) {
```

Comment:
I think I would prefer re-using the `SearchMachines` function instead of creating further functions for that. Maybe you need to extend the search query, but this would also be beneficial for other clients.